### ISSUE TITLE
snippets extensions don't works on Insiders build 1.27

### ISSUE THREAD

**Issue by talentedjerkkk23** [Microsoft/vscode#56403]
### Environment
- VSCode Version: 1.27 Insiders
- OS Version: Windows 7 x64

### Steps to Reproduce:

1. Install VSCode 1.27 Insiders build
2. Install any snippets extension

### Expected Behavior
 auto-completion from the installed extension(-s)  should be shown on typing

### Actual Behavior
auto-completion from the installed extension(-s) are not shown. Only default IntelliSense completions  are shown.

### Additional Info
1. In dev-tools for every snippets extension here the same error message with only different path and name: 
`Expected `contributes.jest-snippets.path` (/c:/Users/user/.vscode-insiders/extensions/andys8.jest-snippets-1.4.0/snippets/snippets.json) to be included inside extension's folder (/C:/Users/user/.vscode-insiders/extensions/andys8.jest-snippets-1.4.0). This might make the extension non-portable.`
and another one:
`/C:/Users/user/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:3230 [ms-python.python]: Expected `contributes.python.path` (/c:/Users/user/.vscode-insiders/extensions/ms-python.python-2018.7.1/snippets/python.json) to be included inside extension's folder (/C:/Users/user/.vscode-insiders/extensions/ms-python.python-2018.7.1). This might make the extension non-portable.`

2. On VSCode 1.26 Stable with absolutely same configuration everything works properly

**Comment by t7yang** [comment]
same here

**Comment by Eventyret** [comment]
To confirm, I have the same here, tested my own plugin and other snippets plugins.

**Comment by sandy081** [comment]
I assume this is caused by the changes in https://github.com/Microsoft/vscode/blob/master/src/vs/base/common/resources.ts#L90

**Comment by jrieken** [comment]
Yeah, the problem is that `resource#joinPath` normalises the windows drive letter. Then it used that to create a new URI. Later, we use `uri.path.indexOf` which is questionable but seems to have worked before. 

@aeschli Never use `fsPath` ;-) I see two fixed here. 

1. `joinPath` function shouldn't have the casing of paths. don't use `fsPath` but build your own equivalent without case changing
2. use `isEqualOrParent` instead of `indexOf`

**Comment by aeschli** [comment]
@jrieken I tried do it without fsPath, but then we loose the getRoot smartness and `normalize` can result in invalid paths.
The windows drive normalization is a challange. I guess there are very strong reasons for it..?

IMO replacing `uri.path.indexOf` with `isEqualOrParent` is really the right fix here.
